novawallet/Modules/Staking/NominationPools/BondMore/Base/NominationPoolBondMoreBasePresenter.swift

```diff
@@ -0,0 +1,242 @@
+            poolBalance: stakingLedger.active
+        )
+    }
+
+    func refreshFee() {
+        guard
+            let stakingLedger = stakingLedger,
+            let bondedPool = bondedPool else {
+            return
+        }
+
+        let inputAmount = getInputAmountInPlank() ?? 0
+
+        fee = nil
+
+        provideFee()
+
+        let points = NominationPools.balanceToPoints(
+            for: inputAmount,
+            totalPoints: bondedPool.points,
+            poolBalance: stakingLedger.active
+        )
+
+        baseInteractor.estimateFee(for: points)
+    }
+
+    func spendingAmount() -> Decimal? {
+        if let inputAmount = getInputAmount(),
+           let fee = fee?.decimal(precision: chainAsset.asset.precision) {
+            return inputAmount + fee
+        } else {
+            return nil
+        }
+    }
+
+    func spendingAmountInPlank() -> BigUInt? {
+        if let inputAmount = getInputAmountInPlank(),
+           let fee = fee {
+            return inputAmount + fee
+        } else {
+            return nil
+        }
+    }
+
+    func getValidations() -> [DataValidating] {
+        let baseValidators = [
+            dataValidatorFactory.hasInPlank(
+                fee: fee,
+                locale: selectedLocale,
+                precision: chainAsset.assetDisplayInfo.assetPrecision
+            ) { [weak self] in
+                self?.refreshFee()
+            },
+            dataValidatorFactory.canSpendAmountInPlank(
+                balance: assetBalance?.transferable,
+                spendingAmount: spendingAmount(),
+                asset: chainAsset.assetDisplayInfo,
+                locale: selectedLocale
+            ),
+            dataValidatorFactory.exsitentialDepositIsNotViolated(
+                spendingAmount: spendingAmountInPlank(),
+                totalAmount: assetBalance?.totalInPlank,
+                minimumBalance: assetBalanceExistance?.minBalance,
+                locale: selectedLocale
+            )
+        ]
+
+        let poolValidators = [
+            dataValidatorFactory.nominationPoolIsDestroing(
+                pool: bondedPool,
+                locale: selectedLocale
+            ),
+            dataValidatorFactory.nominationPoolIsFullyUnbonding(
+                poolMember: poolMember,
+                locale: selectedLocale
+            )
+        ]
+
+        return baseValidators + poolValidators
+    }
+
+    // MARK: - NominationPoolBondMoreBaseInteractorOutputProtocol
+
+    func didReceive(assetBalance: AssetBalance?) {
+        self.assetBalance = assetBalance
+    }
+
+    func didReceive(poolMember: NominationPools.PoolMember?) {
+        let shouldRefreshFee = poolMember?.points != self.poolMember?.points
+
+        self.poolMember = poolMember
+        provideHints()
+
+        if shouldRefreshFee {
+            refreshFee()
+        }
+    }
+
+    func didReceive(bondedPool: NominationPools.BondedPool?) {
```

Comment:
As you don't need to convert balance to point you don't need bonded pool elsewhere

Reply:
it's needed for validation (NPoolIsNotDestroing)